armory/data/datasets.py

```diff
@@ -678,19 +679,52 @@ def _cache_dataset(dataset_dir: str, name: str, subpath: str):
 }
 
 
-def download_all():
+def download_all(dataset_names):
     """
-    Download all datasets to cache.
+    Download all datasets for a scenario or requested datset to cache.
     """
-    errors = []
-    for name, func in SUPPORTED_DATASETS.items():
-        logger.info(f"Downloading (if necessary) dataset {name}")
-        try:
-            func()
-        except Exception:
```

Comment:
We may want (for testing purposes) to be able to call this with no arguments, and have it actually go and download all datasets.